lib/get-posts.js

```diff
@@ -0,0 +1,48 @@
+import dayjs from "dayjs"
+import { paramCase } from "param-case"
+
+import notion from "lib/notion"
+
+const getPosts = async () => {
+  const databaseId = process.env.NOTION_DATABASE_ID
+
+  const database = await notion.databases.query({
+    database_id: databaseId,
+    sorts: [
+      {
+        property: "Date",
+        direction: "descending",
+      },
+    ],
+  })
+
+  const rawPages = database.results
+
+  const pagePromises = rawPages.map(async ({ id }) => {
+    const page = await notion.pages.retrieve({ page_id: id })
+
+    return {
+      slug: `${paramCase(
+        page.properties["Name"][page.properties["Name"].type][0]?.plain_text,
+      )}--${page.id}`,
+      title:
+        page.properties["Name"][page.properties["Name"].type][0]?.plain_text,
+      description:
+        page.properties["Description"][page.properties["Description"].type][0]
+          ?.plain_text || null,
+      date: dayjs(
+        page.properties["Date"][page.properties["Date"].type].start,
+      ).format("DD MMMM YYYY"),
+      cover: page?.cover?.[page.cover.type].url || null,
+      redirect:
+        page.properties["Redirect"][page.properties["Redirect"].type][0]
+          ?.plain_text || null,
+    }
+  })
```

Comment:
```suggestion
  const pagePromises = rawPages.map(({ id }) => {
    const page = await notion.pages.retrieve({ page_id: id })

    return {
      slug: `${paramCase(
        page.properties["Name"][page.properties["Name"].type][0]?.plain_text,
      )}--${page.id}`,
      title:
        page.properties["Name"][page.properties["Name"].type][0]?.plain_text,
      description:
        page.properties["Description"][page.properties["Description"].type][0]
          ?.plain_text || null,
      date: dayjs(
        page.properties["Date"][page.properties["Date"].type].start,
      ).format("DD MMMM YYYY"),
      cover: page?.cover?.[page.cover.type].url || null,
      redirect:
        page.properties["Redirect"][page.properties["Redirect"].type][0]
          ?.plain_text || null,
    }
  })
```

reutilise les helpers du commentaire d'avant